From aab3b95d63a310d4d45fe5c348e2605ecbf43692 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 3 Aug 2023 20:43:06 -0500
Subject: [PATCH] Solus: Separate unit stop timeout out and set it to 10s by
 default

default-timeout-sec and default-user-timeout-sec both set many timeouts throughout systemd. While these 
timeouts are important we want to set the unit stop timeout to a lower value so that logging out and shutdowns/restarts
completely quickly.

Add a separate constant for this, set it to 10 seconds and then use that for the unit stop timeout only.

---
 src/basic/constants.h   | 1 +
 src/core/main.c         | 2 +-
 src/core/manager.c      | 2 +-
 src/core/system.conf.in | 2 +-
 src/core/user.conf.in   | 2 +-
 5 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/basic/constants.h b/src/basic/constants.h
index 5d68cc6332..4221dd4029 100644
--- a/src/basic/constants.h
+++ b/src/basic/constants.h
@@ -29,6 +29,7 @@
 
 /* Many different things, but also system unit start/stop */
 #define DEFAULT_TIMEOUT_USEC (DEFAULT_TIMEOUT_SEC*USEC_PER_SEC)
+#define DEFAULT_STOP_TIMEOUT_USEC (10*USEC_PER_SEC)
 /* User unit start/stop */
 #define DEFAULT_USER_TIMEOUT_USEC (DEFAULT_USER_TIMEOUT_SEC*USEC_PER_SEC)
 /* Timeout for user confirmation on the console */
diff --git a/src/core/main.c b/src/core/main.c
index 299cafeda8..c973cc65c6 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -2452,7 +2452,7 @@ static void reset_arguments(void) {
         arg_default_std_error = EXEC_OUTPUT_INHERIT;
         arg_default_restart_usec = DEFAULT_RESTART_USEC;
         arg_default_timeout_start_usec = manager_default_timeout(arg_system);
-        arg_default_timeout_stop_usec = manager_default_timeout(arg_system);
+        arg_default_timeout_stop_usec = DEFAULT_STOP_TIMEOUT_USEC;
         arg_default_timeout_abort_usec = manager_default_timeout(arg_system);
         arg_default_timeout_abort_set = false;
         arg_default_device_timeout_usec = manager_default_timeout(arg_system);
diff --git a/src/core/manager.c b/src/core/manager.c
index 1853526e3f..79e6e3e5d1 100644
--- a/src/core/manager.c
+++ b/src/core/manager.c
@@ -836,7 +836,7 @@ int manager_new(LookupScope scope, ManagerTestRunFlags test_run_flags, Manager *
                 .default_tasks_accounting = true,
                 .default_tasks_max = TASKS_MAX_UNSET,
                 .default_timeout_start_usec = manager_default_timeout(scope == LOOKUP_SCOPE_SYSTEM),
-                .default_timeout_stop_usec = manager_default_timeout(scope == LOOKUP_SCOPE_SYSTEM),
+                .default_timeout_stop_usec = DEFAULT_STOP_TIMEOUT_USEC,
                 .default_restart_usec = DEFAULT_RESTART_USEC,
                 .default_device_timeout_usec = manager_default_timeout(scope == LOOKUP_SCOPE_SYSTEM),
 
diff --git a/src/core/system.conf.in b/src/core/system.conf.in
index 9572b57f17..cfeb9cf0ef 100644
--- a/src/core/system.conf.in
+++ b/src/core/system.conf.in
@@ -44,7 +44,7 @@
 #DefaultStandardOutput=journal
 #DefaultStandardError=inherit
 #DefaultTimeoutStartSec={{DEFAULT_TIMEOUT_SEC}}s
-#DefaultTimeoutStopSec={{DEFAULT_TIMEOUT_SEC}}s
+#DefaultTimeoutStopSec=10s
 #DefaultTimeoutAbortSec=
 #DefaultDeviceTimeoutSec={{DEFAULT_TIMEOUT_SEC}}s
 #DefaultRestartSec=100ms
diff --git a/src/core/user.conf.in b/src/core/user.conf.in
index d67650b858..c34b8eba52 100644
--- a/src/core/user.conf.in
+++ b/src/core/user.conf.in
@@ -25,7 +25,7 @@
 #DefaultStandardOutput=inherit
 #DefaultStandardError=inherit
 #DefaultTimeoutStartSec={{DEFAULT_USER_TIMEOUT_SEC}}s
-#DefaultTimeoutStopSec={{DEFAULT_USER_TIMEOUT_SEC}}s
+#DefaultTimeoutStopSec=10s
 #DefaultTimeoutAbortSec=
 #DefaultDeviceTimeoutSec={{DEFAULT_USER_TIMEOUT_SEC}}s
 #DefaultRestartSec=100ms
-- 
2.41.0

